### ISSUE TITLE
redundantFileprivate breaks public protocol conformances in (file)private types

### ISSUE THREAD

**Issue by ChristopherRogers** [nicklockwood/SwiftFormat#391]
If you have a (file)private type with a conformance to a public protocol, redundantFileprivate will break the conformance. This is because the requirements will be hidden with `private`.

Example:

```swift
private class MyClass: Equatable {
    fileprivate static func == (_: MyClass, _: MyClass) -> Bool {
        return true
    }
}
```

...will get transformed into...

```swift
private class MyClass: Equatable {
    private static func == (_: MyClass, _: MyClass) -> Bool {
        return true
    }
}
```

**Comment by nicklockwood** [comment]
@christopherrogers where is Myclass == being accessed from - is it internally within the class, or from somewhere else in the same file?

**Comment by ChristopherRogers** [comment]
This was just a self-contained example I wrote, so it's not being used anywhere. It's just required to be fileprivate in order for it to be at the same visibility of the class.

**Comment by ChristopherRogers** [comment]
(Otherwise it won't compile.)

**Comment by nicklockwood** [comment]
@christopherrogers ah, I see.

**Comment by nicklockwood** [comment]
@ChristopherRogers fixed in 0.40.3.
